sensor/tests/resource/service/service_test.go

```diff
@@ -450,66 +477,183 @@ func (s *DeploymentExposureSuite) Test_NodePortPermutationWithPod() {
+					return err
+				}
+				setDynamicFieldsInSlice(c.orderedResources, c.portConfig, serviceNodePortFmt, getPort(s.T()), c.selector, setNodePort, setPortConfigNode)
+				return nil
+			}),
 		)
 	}
 }
+
+var nextPort int32 = 30000
+
+const (
+	MaxPort = 30100
+)
+
+func getPort(t *testing.T) int32 {
+	if nextPort > MaxPort {
+		// If we reached the maximum usable port we fail the test
+		t.Fatalf("Reached maximum usable port:\nMaxPort = %d, current port = %d", MaxPort, nextPort)
+	}
+	ret := nextPort
+	nextPort++
+	return ret
+}
+
+type serviceFunc func(*v1.Service, string, int32, map[string]string)
+
+type portConfigFunc func([]*storage.PortConfig, string, int32)
+
+func setNodePort(svc *v1.Service, name string, port int32, sel map[string]string) {
+	svc.ObjectMeta = metav1.ObjectMeta{
+		Name:      fmt.Sprintf(name, port),
+		Namespace: resource.DefaultNamespace,
+	}
+	svc.Spec = v1.ServiceSpec{
+		Type:     v1.ServiceTypeNodePort,
+		Selector: sel,
+		Ports: []v1.ServicePort{
+			{
+				Port: 80,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 80,
+				},
+				NodePort: port,
+			},
+		},
+	}
+}
+
+func setLoadBalancer(svc *v1.Service, name string, port int32, sel map[string]string) {
+	svc.ObjectMeta = metav1.ObjectMeta{
+		Name:      fmt.Sprintf(name, port),
+		Namespace: resource.DefaultNamespace,
+	}
+	svc.Spec = v1.ServiceSpec{
+		Type:     v1.ServiceTypeLoadBalancer,
+		Selector: sel,
+		Ports: []v1.ServicePort{
+			{
+				Port: 80,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 80,
+				},
+				NodePort: port,
+			},
+		},
+	}
+}
+
+func setPortConfigExternal(portConfig []*storage.PortConfig, name string, port int32) {
+	if len(portConfig) != 1 {
+		return
+	}
+	portConfig[0] = &storage.PortConfig{
+		Protocol:      "TCP",
+		ContainerPort: 80,
+		Exposure:      storage.PortConfig_EXTERNAL,
+		ExposureInfos: []*storage.PortConfig_ExposureInfo{
+			{
+				ServiceName: fmt.Sprintf(name, port),
+				ServicePort: 80,
+				NodePort:    port,
+				Level:       storage.PortConfig_EXTERNAL,
+			},
+		},
+	}
+}
+
+func setPortConfigNode(portConfig []*storage.PortConfig, name string, port int32) {
+	if len(portConfig) != 1 {
+		return
+	}
+	portConfig[0] = &storage.PortConfig{
+		Protocol:      "TCP",
+		ContainerPort: 80,
+		Exposure:      storage.PortConfig_NODE,
+		ExposureInfos: []*storage.PortConfig_ExposureInfo{
+			{
+				ServiceName: fmt.Sprintf(name, port),
+				ServicePort: 80,
+				NodePort:    port,
+				Level:       storage.PortConfig_NODE,
+			},
+		},
+	}
+}
+
+func setDynamicFields(svc *v1.Service, name string, port int32, sel map[string]string, serviceFunc func(*v1.Service, string, int32, map[string]string)) {
+	serviceFunc(svc, name, port, sel)
+}
+
+func setDynamicFieldsInSlice(resources []resource.K8sResourceInfo, portConfig []*storage.PortConfig, name string, port int32, sel map[string]string, svcFunc serviceFunc, pcFunc portConfigFunc) {
```

Comment:
I think naming can be improved here. I suggest using the suffix `fn` for functions passed as parameter. 

Also, the parameter name `pcFunc` is a bit cryptic, could we call it as it is, something like: `portConfigFn`? Same for `svcFunc`.

Reply:
Done